node/risk-app/server/routes/map.js

```diff
@@ -12,7 +12,9 @@ module.exports = {
       const { easting, northing } = query
       const address = request.yar.get('address')
       const path = request.path
-      const backLinkUri = defineBackLink(path)
+      const previousPage = request.yar.get('previousPage')
+      console.log('Retrieved previousPage for /map:', previousPage)
+      const backLinkUri = defineBackLink(path, previousPage)
```

Comment:
Something about this does not seem to be working correctly. When you log the previous page when going to the map from view a map showing areas at risk of flooding](http://localhost:3000/map) it shows as R&S. Its likely going back to the correct page as there isnt a postcode so it selects postcode as default. Might want to check this and write a test that covers it.